### DISCUSSION TITLE
useNavigation (formerly useTransition) state is always `idle`

### DISCUSSION THREAD

**Discussion by loolooii** [remix-run/remix#9683] [Q&A]
I'm trying to show a loading bar whenever the page is still loading (whether accessing directly or after clicking a link), but I useNavigation() always has an `idle` state. It's never `loading`. 

Basically I have a loader, which fetches some data and in my component (root.tsx) I'm doing this:

```
const data = useLoaderData<typeof loader>();
const navigation = useNavigation();
console.log('state: ', navigation.state) // this is always idle
```

What can be the problem here?

**Comment by GabenGar** [comment]
Does it change if you instead `JSON.stringify()` the value?
